mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulPrepareGameScreen.kt

```diff
@@ -27,8 +20,51 @@ class PrepareGameScreenStateImpl(private val user: AuthenticatedUser) : PrepareG
 @Composable
 fun StatefulPrepareGameScreen(
     user: AuthenticatedUser,
+    navigateToGame: (match: Match) -> Unit,
+    onCancelClick: () -> Unit,
     modifier: Modifier = Modifier,
 ) {
-  val state = remember(user) { PrepareGameScreenStateImpl(user) }
-  PrepareGameScreen(state, modifier)
+  val scope = rememberCoroutineScope()
+  val chessFacade = LocalChessFacade.current
+  val opponents =
+      remember(user) { user.following }.collectAsState(emptyList()).value.map { ProfileAdapter(it) }
+
+  val state =
+      remember(user, opponents, navigateToGame, onCancelClick, scope, chessFacade) {
+        SnapshotPrepareGameScreenState(
+            user, opponents, navigateToGame, onCancelClick, scope, chessFacade)
+      }
+  PrepareGameScreen(state = state, modifier = modifier, key = { it.uid })
```

Comment:
I suggest doing as much work as possible in the ViewModel class directly, similarly to this : https://github.com/epfl-SDP/android/blob/main/mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulFollowingScreen.kt#L128-L138

It makes code that's easier to test, and helps avoid some unnecessary recompositions (for instance, if the `opponents` list changes because an opponent changed their profile name for instance, a new `SnapshotPrepareGameScreenState` and the current selection will be lost).

Reply:
Fixed in [5fb0802](https://github.com/epfl-SDP/android/pull/200/commits/5fb08021c954bdad0ad44de1e9e111b7a5eff163) (I think?)